core/src/main/java/org/mule/transformer/simple/ObjectToString.java

```diff
@@ -85,6 +85,7 @@ protected String createStringFromInputStream(InputStream input, String outputEnc
             try
             {
                 input.close();
+                input.reset();
```

Comment:
i don't fully understand what you're saying, but it sounds like the solution doesn't work without this, so I guess that revert it is.